src/odemis/driver/picoquant.py

```diff
@@ -1069,7 +2407,184 @@ def PH_GetElapsedMeasTime(self, i, p_elapsed):
+        ndbuffer[...] = numpy.random.randint(0, maxval + 1, PH_HISTCHAN).astype(
+            numpy.uint32
+        )
+
+
+class FakeHHDLL(object):
+    """
+    Fake HHDLL. It basically simulates one connected device, which returns
+    reasonable values.
+    """
+
+    def __init__(self):
+        self._idx = 0
+        self._mode = None
+        self._refsource = 0
+        self._sn = b"10234567"
+        self._base_res = 4  # ps
+        self._bincode = 0  # binning power
+        self._numinput = 2
+        self._inputLevel = []
+        self._inputZc = []
+        self._inputOffset = []
+        self._syncRate = 50000
+        self._syncPeriod = 2000.0
+
+        # start/ (expected) end time of the current acquisition (or None if not started)
+        self._acq_start = None
+        self._acq_end = None
+        self._last_acq_dur = None  # s
+
+    # General Functions
+    # These functions work independent from any device.
+
+    def HH_GetErrorString(self, errstring, errcode):
+        errstring.value = b"Fake error string"
+
+    def HH_GetLibraryVersion(self, ver_str):
+        ver_str.value = b"3.00"
+
+    # Device Specific Functions
+    # All functions below are device specific and require a device index.
+
+    def HH_OpenDevice(self, i, sn_str):
+        if i == self._idx:
+            sn_str.value = self._sn
+        else:
+            raise HHError(-1, HHDLL.err_code[-1])  # ERROR_DEVICE_OPEN_FAIL
+
+    def HH_CloseDevice(self, i):
+        self._mode = None
+
+    def HH_Initialize(self, i, mode, refsource):
+        self._mode = mode
+        self._refsource = refsource
+
+    # Functions for Use on Initialized Devices
+    # All functions below can only be used after HH_Initialize was successfully called.
+
+    def HH_GetHardwareInfo(self, i, mod, partnum, ver):
+        mod.value = b"FakeHarp 400"
+        partnum.value = b"12345"
+        ver.value = b"2.0"
+
+    def HH_GetFeatures(self, i, features):
+        # Not needed?
+        pass
+
+    def HH_GetSerialNumber(self, i, sn_str):
+        sn_str.value = self._sn
+
+    def HH_GetBaseResolution(self, i, p_resolution, p_binsteps):
+        resolution = _deref(p_resolution, c_double)
+        binsteps = _deref(p_binsteps, c_int)
+        resolution.value = self._base_res
+        binsteps.value = self._bincode
+
+    def HH_GetNumOfInputChannels(self, i, p_nchannels):
+        nchannels = _deref(p_nchannels, c_int)
+        nchannels.value = self._numinput
+
+    def HH_GetNumOfModules(self, i, nummod):
+        # Not needed?
+        pass
+
+    def HH_GetModuleInfo(self, i, modidx, modelcode, versioncode):
+        # Not needed?
+        pass
+
+    def HH_GetModuleIndex(self, i, channel, modidx):
+        # Not needed?
+        pass
+
+    def HH_GetHardwareDebugInfo(self, i, debuginfo):
+        debuginfo.value = b"Fake hardware debug info"
+        # Not needed?
+
+    def HH_Calibrate(self, i):
+        pass
```

Comment:
Shouldn't be there some type of logging saying "not implemented function nameOfFunction is called" ?